instrumentation/apache-httpclient/apache-httpclient-5.0/src/test/java/io/opentelemetry/javaagent/instrumentation/apachehttpclient/v5_0/ConnectionTest.java

```diff
@@ -0,0 +1,203 @@
+      try (CloseableHttpClient client = HttpClients.createDefault()) {
+        final ClassicHttpRequest httpGet = ClassicRequestBuilder.get(mockServer.url(urlPath))
+            .addHeader("Content-Type", "application/json").build();
+        client.execute(httpGet, response -> null);
+      }
+    });
+
+    assertThat(instrumentation.waitForTraces(1))
+        .hasSize(1)
+        .hasTracesSatisfyingExactly(
+            trace ->
+              trace
+                  .hasSize(1)
+                  .hasSpansSatisfyingExactly(
+                      span -> span
+                          .hasName("GET")
+                          .hasKind(SpanKind.CLIENT)
+                          .hasAttribute(AttributeKey.stringKey("http.method"), "GET")
+                          .hasAttribute(AttributeKey.stringArrayKey("http.request.header.content_type"),
+                              List.of("application/json"))
+                          .hasAttribute(AttributeKey.stringKey("http.request.body"), "null")
+                          .hasException(thrown)
+                          .hasStatus(StatusData.error())
+                  ));
+  }
+
+  @Test
+  void testMalformedChunk() {
+    final String urlPath = "/malformed";
+
+    mockServer.stubFor(get(urlPathEqualTo(urlPath))
+        .willReturn(aResponse()
+            .withFault(Fault.MALFORMED_RESPONSE_CHUNK)
+        )
+    );
+
+    try (CloseableHttpClient client = HttpClients.createDefault()) {
+      final ClassicHttpRequest httpGet = ClassicRequestBuilder.get(mockServer.url(urlPath))
+          .addHeader("Content-Type", "application/json").build();
+      client.execute(httpGet, response -> null);
+    } catch (MalformedChunkCodingException mcce) {
+      // Expecting this one
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    assertThat(instrumentation.waitForTraces(1))
+        .hasSize(1)
+        .hasTracesSatisfyingExactly(
+            trace ->
+                trace
+                    .hasSize(1)
+                    .hasSpansSatisfyingExactly(
+                        span -> span
+                            .hasName("GET")
+                            .hasKind(SpanKind.CLIENT)
+                            .hasAttribute(AttributeKey.stringKey("http.method"), "GET")
+                            .hasAttribute(AttributeKey.stringArrayKey("http.request.header.content_type"),
+                                List.of("application/json"))
+                            .hasAttribute(AttributeKey.stringKey("http.request.body"), "null")
+                            .hasAttribute(AttributeKey.stringKey("http.response.body"), "")
+                            .hasAttribute(AttributeKey.stringArrayKey("http.response.header.transfer_encoding"),
+                                List.of("chunked"))
+                            .hasAttribute(AttributeKey.longKey("http.status_code"), 200L)
+                            .hasStatus(StatusData.unset())
```

Comment:
This is not great for the instrumentation to do :-)
Status should be `OK`.

Reply:
This is not us, this is what OTeL sets it to in this situation

Reply:
Yes and I mildly disapprove of it